src/main/java/it/gov/pagopa/reportingorgsenrollment/service/EnrollmentsServiceTableImpl.java

```diff
@@ -0,0 +1,145 @@
+import io.micronaut.core.convert.TypeConverter;
+import it.gov.pagopa.reportingorgsenrollment.entity.OrganizationEntity;
+import it.gov.pagopa.reportingorgsenrollment.exception.AppError;
+import it.gov.pagopa.reportingorgsenrollment.exception.AppException;
+import it.gov.pagopa.reportingorgsenrollment.model.response.OrganizationModelResponse;
+import jakarta.inject.Singleton;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Primary
+@Singleton
+@Slf4j
+public class EnrollmentsServiceTableImpl implements EnrollmentsService {
+
+    public final static String FISCAL_CODE_PROPERTY = "organizationFiscalCode";
+    public final static String ON_BOARDING_DATE_PROPERTY = "organizationOnBoardingDate";
+
+    private String connectionString;
+    private String organizationsTable;
+
+    public EnrollmentsServiceTableImpl(@Value("${tables.reporting.connection}") String connectionString,
+                                       @Value("${tables.reporting.organizations.table}") String organizationsTable) {
+        this.connectionString = connectionString;
+        this.organizationsTable = organizationsTable;
+    }
+
+    @Override
+    public OrganizationModelResponse createOrganization(String organizationFiscalCode) {
+        OrganizationEntity organizationEntity = new OrganizationEntity(organizationFiscalCode, LocalDateTime.now());
+
+        try {
+            TableClient tableClient = getOrganizationTable();
+
+            TableEntity entity = new TableEntity(organizationFiscalCode, organizationFiscalCode)
+                    .addProperty(FISCAL_CODE_PROPERTY, organizationEntity.getOrganizationFiscalCode())
+                    .addProperty(ON_BOARDING_DATE_PROPERTY, organizationEntity.getOrganizationOnBoardingDate().toString());
+            tableClient.createEntity(entity);
+
+            return organizationResponseTypeConverter().convert(organizationEntity, OrganizationModelResponse.class).orElse(null);
+        }
+        catch (TableServiceException e) {
+            log.error("Error in processing create organization", e);
+            if (e.getResponse().getStatusCode() == AppError.ORGANIZATION_DUPLICATED.httpStatus.getCode())
+                throw new AppException(AppError.ORGANIZATION_DUPLICATED, organizationFiscalCode);
+            else
+                throw new AppException(AppError.INTERNAL_ERROR, organizationFiscalCode);
+        }
+    }
+
+    @Override
+    public void removeOrganization(String organizationFiscalCode) {
+        try {
+            TableClient tableClient = getOrganizationTable();
+
+            // getEntity to be API compliant (delete method is void, but it needs to return 404 if not found: handled by exception)
+            tableClient.getEntity(organizationFiscalCode, organizationFiscalCode);
+            tableClient.deleteEntity(organizationFiscalCode, organizationFiscalCode);
```

Comment:
if not exist entity `deleteEntity` not catch this case ?

Reply:
Exactly, without a prior `getEntity` it would return `200`